xiblint/rules/color_assets.py

```diff
@@ -0,0 +1,80 @@
+from xiblint.rules import Rule
+from xiblint.xibcontext import XibContext
+import glob
+import json
+
+class ColorAssets(Rule):
+    def __init__(self, config):
+        asset_catalog_path = config.get('asset_catalog', None)
+        if asset_catalog_path is None:
+            # TODO: Proper error
+            print("Asset catalog not found. Please configure 'asset_catalog'.")
+            return
+
+        self.assets = glob.glob("{}/**/*.colorset".format(asset_catalog_path))
+
+        if not self.assets:
+            # TODO: Proper error
+            print("Failed to load asset catalog at: '{}'".format(asset_catalog_path))
+            return
+
+        self.colors = {}
+
+    """
+    Ensures cached named colors match the named colors from an asset catalog.
+
+    Example configuration:
+    {
+      "asset_catalog": "Resources/Color.xcassets"
+    }
+    """
+    def check(self, context):  # type: (XibContext) -> None
+        for element in context.tree.findall(".//namedColor"):
+            name = element.get("name")
+            if name is None:
+                context.error(element, "Named color is missing a name.")
+                continue
+
+            color = self.load_color(name)
+            if color is None:
+                context.error(element, "Named color '{}' is missing from asset catalog.".format(name))
+                continue
+
+            color_element = element.find("color")
+            if color_element is None:
```

Comment:
```suggestion
            if not color_element:
```

Reply:
I had to switch this back to `is None`. It gave me a warning using `not` saying that was deprecated and I should switch to something with `len()` instead. Checking for nil seemed better than checking the length 🤷‍♂